### DISCUSSION TITLE
Scheduled task notifications don't go away, tasks are suspected of not running at all

### DISCUSSION THREAD

**Discussion by chuwza** [d4rken-org/sdmaid-se#834] [Q&A] [answered]
Hi, Thanks for your work, SD Maid SE is good, but I'm still finding problem with scheduled tasks.

I posed this question #828 before. Now that I've updated the latest version 0.17.1, a notification does appear when I schedule a task, but this notification doesn't go away after two hours (I manually cleaned the trash the whole time in only 30 minutes)
![Screenshot_20231129-125125_Lawnchair (小)](https://github.com/d4rken-org/sdmaid-se/assets/30049891/ce2f893c-a252-42f1-b76b-b52d19c7bc5c)

and going into SD Maid SE I find that there aren't any running tasks.
![Screenshot_20231129-125150_SD_Maid_SE (小)](https://github.com/d4rken-org/sdmaid-se/assets/30049891/88784ea3-da24-41bb-a2cf-d9e0f9d621c5)

So this notification simply makes the previous flickering notification resident? And never goes away?

One of my biggest concerns is: are the scheduled tasks actually running? Because when I go into SD Maid SE, I get no indication that the task is running

**Comment by d4rken** [comment]
There are 3 types of notifications that can be visible.

* Task activity notifications, these will display when SD Maid is running any tasks (no matter if manual or via scheduler). Only displayed while this is on-going.
* Scheduler activity notifications, these will display while tasks triggered by the scheduler are being executed. Only displayed while this is on-going.
* Scheduler result notifications (your screenshot). These are displayed after scheduler execution finished.

In your case, the notification signals that a scheduled operation has successfully finished 2 hours ago. Opening SD Maid will not show anything because it's 2 hours later.

**Comment by chuwza** [comment]
> There are 3 types of notifications that can be visible.
> 
> * Task activity notifications, these will display when SD Maid is running any tasks (no matter if manual or via scheduler). Only displayed while this is on-going.
> * Scheduler activity notifications, these will display while tasks triggered by the scheduler are being executed. Only displayed while this is on-going.
> * Scheduler result notifications (your screenshot). These are displayed after scheduler execution finished.
> 
> In your case, the notification signals that a scheduled operation has successfully finished 2 hours ago. Opening SD Maid will not show anything because it's 2 hours later.

"In your case, the notification signals that a scheduled operation has successfully finished 2 hours ago. Opening SD Maid will not show anything because it's 2 hours later."

If that's what's being said, then it's more likely that there's something wrong with the program mission. Because 2 hours ago was the start time of my scheduled task, which means that the scheduled task only used a few seconds to complete. And in fact, I've tried scanning but not cleaning up the junk, and I've found that it takes about 30 minutes to do so.

I have now tested it again and all I can determine is that there may be a problem with the scheduled task, see my YouTube video(https://youtu.be/8Re9MbuwkrQ). I set the time for the scheduled task to be 20:10, and after that time the "Scheduler activity notifications" appeared for a moment, and then the "Scheduler result notifications" appeared again for a a moment. There is no way that the garbage cleanup task can be completed within a few seconds. 

(I've tested the garbage cleanup time before and it takes about 30 minutes, I didn't turn on one-click cleanup to keep the test, and I chose to cancel it directly after the cleanup result came out, without cleaning the garbage. So there's no way this garbage cleanup will only take 1 second.)

**Comment by d4rken** [comment]
> I have now tested it again and all I can determine is that there may be a problem with the scheduled task, see my YouTube video(https://youtu.be/8Re9MbuwkrQ). I set the time for the scheduled task to be 20:10, and after that time the "Scheduler activity notifications" appeared for a moment, and then the "Scheduler result notifications" appeared again for a a moment. There is no way that the garbage cleanup task can be completed within a few seconds.

That's quick. Note that the accessibility actions don't run via scheduler, but even without that it's a bit too quick. I'm also suprised to not see the other notifications, though that might be because it was too quick.

Can you record a debug log of what the video showed, then we can see what is executed.

https://github.com/d4rken-org/sdmaid-se/wiki/Bugs#debug-log

> (I've tested the garbage cleanup time before and it takes about 30 minutes, I didn't turn on one-click cleanup to keep the test, and I chose to cancel it directly after the cleanup result came out, without cleaning the garbage. So there's no way this garbage cleanup will only take 1 second.)

If you want to play around with that, you can enable debug mode, and then on the debug card enable "Dry run" it will only simulate deletion so tests can be repeated.

**Comment by chuwza** [comment]
> > I have now tested it again and all I can determine is that there may be a problem with the scheduled task, see my YouTube video(https://youtu.be/8Re9MbuwkrQ). I set the time for the scheduled task to be 20:10, and after that time the "Scheduler activity notifications" appeared for a moment, and then the "Scheduler result notifications" appeared again for a a moment. There is no way that the garbage cleanup task can be completed within a few seconds.
> 
> That's quick. Note that the accessibility actions don't run via scheduler, but even without that it's a bit too quick. I'm also suprised to not see the other notifications, though that might be because it was too quick.
> 
> Can you record a debug log of what the video showed, then we can see what is executed.
> 
> https://github.com/d4rken-org/sdmaid-se/wiki/Bugs#debug-log
> 
> > (I've tested the garbage cleanup time before and it takes about 30 minutes, I didn't turn on one-click cleanup to keep the test, and I chose to cancel it directly after the cleanup result came out, without cleaning the garbage. So there's no way this garbage cleanup will only take 1 second.)
> 
> If you want to play around with that, you can enable debug mode, and then on the debug card enable "Dry run" it will only simulate deletion so tests can be repeated.

Hi, I run the test, here's the YouTube link (https://youtu.be/ORpSUWSs4Zg) (I'm afraid my steps are wrong, so I recorded a video) and here's the log file
[eu.darken.sdmse_logfile_1701261079893.zip](https://github.com/d4rken-org/sdmaid-se/files/13500551/eu.darken.sdmse_logfile_1701261079893.zip)

**Comment by d4rken** [comment]
In that log the tools are all disabled for the schedule:

```useCorpseFinder=false, useSystemCleaner=false, useAppCleaner=false```

This could be why no tasks are being executed?

**Comment by chuwza** [comment]
> In that log the tools are all disabled for the schedule:
> 
> `useCorpseFinder=false, useSystemCleaner=false, useAppCleaner=false`
> 
> This could be why no tasks are being executed?

Got it. So is this a problem with my app setting? Or maybe it's a problem with SD Maid SE?

**Comment by d4rken** [comment, marked as answer]
Well do you have the tools enabled for the schedule?

![Screenshot from 2023-11-29 14-57-29](https://github.com/d4rken-org/sdmaid-se/assets/1439229/59126add-d4c7-4ad6-ac5b-b288a451f283)

**Comment by chuwza** [comment]
> Well do you have the tools enabled for the schedule?
> 
> ![Screenshot from 2023-11-29 14-57-29](https://private-user-images.githubusercontent.com/1439229/286617710-59126add-d4c7-4ad6-ac5b-b288a451f283.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNjY2MTQsIm5iZiI6MTcwMTI2NjMxNCwicGF0aCI6Ii8xNDM5MjI5LzI4NjYxNzcxMC01OTEyNmFkZC1kNGM3LTRhZDYtYWM1Yi1iMjg4YTQ1MWYyODMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjlUMTM1ODM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjMxNjk3MTExNjg4MmVjYzY5NWJiYWRiNzBmZmMzNWRhMzJhZGQyZWMyOWI0YzA2MTdlYjY5ZjM1YzFmODI5YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9t7v8cLdoKDhyOiHm5MJUb4LtqAXmrcYahahW9X_vzQ)

Sorry, that's my problem, I don't have that option turned on. Thank you very much!

**Comment by gabriel20xx** [comment]
> Hi, Thanks for your work, SD Maid SE is good, but I'm still finding problem with scheduled tasks.
> 
> I posed this question #828 before. Now that I've updated the latest version 0.17.1, a notification does appear when I schedule a task, but this notification doesn't go away after two hours (I manually cleaned the trash the whole time in only 30 minutes)
> ![Screenshot_20231129-125125_Lawnchair (小)](https://github.com/d4rken-org/sdmaid-se/assets/30049891/ce2f893c-a252-42f1-b76b-b52d19c7bc5c)
> 
> and going into SD Maid SE I find that there aren't any running tasks.
> ![Screenshot_20231129-125150_SD_Maid_SE (小)](https://github.com/d4rken-org/sdmaid-se/assets/30049891/88784ea3-da24-41bb-a2cf-d9e0f9d621c5)
> 
> So this notification simply makes the previous flickering notification resident? And never goes away?
> 
> One of my biggest concerns is: are the scheduled tasks actually running? Because when I go into SD Maid SE, I get no indication that the task is running

I also have the same issue with the undissmisable persistant scheduler successful notification. Since today (installed new version of sd maid yesterday) i can't dismiss it anymore.

**Reply by d4rken** [reply]
You are right, the OP of #834 had two issues. The notifications don't dismiss, and they didn't enable tasks.

I've created a bug ticket for the first one #838 will be fixed in the next update.
